price-feeder/oracle/provider/mexc.go

```diff
@@ -0,0 +1,504 @@
+package provider
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"net/url"
+	"strconv"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/cosmos/cosmos-sdk/telemetry"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/gorilla/websocket"
+	"github.com/rs/zerolog"
+	"github.com/umee-network/umee/price-feeder/oracle/types"
+)
+
+const (
+	mexcWSHost   = "wbs.mexc.com"
+	mexcWSPath   = "/raw/ws"
+	mexcRestHost = "https://www.mexc.com"
+	mexcRestPath = "/open/api/v2/market/ticker"
+)
+
+var _ Provider = (*MexcProvider)(nil)
+
+type (
+	// MexcProvider defines an Oracle provider implemented by the Mexc public
+	// API.
+	//
+	// REF: https://mxcdevelop.github.io/apidocs/spot_v2_en/#ticker-information
+	// REF: https://mxcdevelop.github.io/apidocs/spot_v2_en/#k-line
+	// REF: https://mxcdevelop.github.io/apidocs/spot_v2_en/#overview
+	MexcProvider struct {
+		wsURL           url.URL
+		wsClient        *websocket.Conn
+		logger          zerolog.Logger
+		mtx             sync.RWMutex
+		endpoints       Endpoint
+		tickers         map[string]types.TickerPrice   // Symbol => TickerPrice
+		candles         map[string][]types.CandlePrice // Symbol => CandlePrice
+		subscribedPairs map[string]types.CurrencyPair  // Symbol => types.CurrencyPair
+	}
+
+	// MexcTickerResponse is the ticker price response object.
+	MexcTickerResponse struct {
+		Symbol map[string]MexcTickerData `json:"data"` // e.x. ATOM_USDT
+	}
+	MexcTickerData struct {
+		LastPrice float64 `json:"p"` // Last price ex.: 0.0025
+		Volume    float64 `json:"v"` // Total traded base asset volume ex.: 1000
+	}
```

Comment:
don't we use similar structure with other providers? Is there anything we can reuse?